From 8620a167275b63205dc40b7e236f893311469929 Mon Sep 17 00:00:00 2001
From: Dhruv Dang <hi@dhruv.io>
Date: Mon, 13 Jul 2020 22:35:55 -0700
Subject: [PATCH 1/2] Support GHC >= 8.8 with MonadFail

Uses conditional statement for the C pre-processor for backward compatibility.
---
 src/HIndent/Types.hs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/HIndent/Types.hs b/src/HIndent/Types.hs
index 78d1ceb..0bfaf68 100644
--- a/src/HIndent/Types.hs
+++ b/src/HIndent/Types.hs
@@ -71,7 +71,11 @@ data Config = Config
     }
 
 -- | Parse an extension.
+#if __GLASGOW_HASKELL__ >= 808
+readExtension :: (Monad m, MonadFail m) => String -> m Extension
+#else
 readExtension :: Monad m => String -> m Extension
+#endif
 readExtension x =
   case classifyExtension x -- Foo
        of

From 75e6f1038dfbbe8ebd9481a5df51586b780a4e74 Mon Sep 17 00:00:00 2001
From: Dhruv Dang <hi@dhruv.io>
Date: Fri, 17 Jul 2020 17:04:21 -0700
Subject: [PATCH 2/2] Add cpp GHC option pragma

---
 src/HIndent/Types.hs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/HIndent/Types.hs b/src/HIndent/Types.hs
index 0bfaf68..18e35a9 100644
--- a/src/HIndent/Types.hs
+++ b/src/HIndent/Types.hs
@@ -1,3 +1,4 @@
+{-# OPTIONS_GHC -cpp #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE GADTs #-}
